mne/export/_edf.py

```diff
@@ -17,71 +17,38 @@
 def _export_raw(fname, raw, physical_range, add_ch_type):
     """Export Raw objects to EDF files.
 
-    TODO: if in future the Info object supports transducer or
-    technician information, allow writing those here.
+    TODO: if in future the Info object supports transducer or technician information,
+    allow writing those here.
     """
-    # get EEG-related data in uV
+    # get voltage-based data in uV
     units = dict(
         eeg="uV", ecog="uV", seeg="uV", eog="uV", ecg="uV", emg="uV", bio="uV", dbs="uV"
     )
 
-    digital_min = -32767
-    digital_max = 32767
+    digital_min, digital_max = -32767, 32767
 
     # load data first
     raw.load_data()
 
-    # remove extra STI channels
-    orig_ch_types = raw.get_channel_types()
-    drop_chs = []
-    if "stim" in orig_ch_types:
-        stim_index = np.argwhere(np.array(orig_ch_types) == "stim")
-        stim_index = np.atleast_1d(stim_index.squeeze()).tolist()
-        drop_chs.extend([raw.ch_names[idx] for idx in stim_index])
-        warn(f"Exporting STIM channels is not supported, dropping indices {stim_index}")
-
-    # Add warning if any channel types are not voltage based.
-    # Users are expected to only export data that is voltage based,
-    # such as EEG, ECoG, sEEG, etc.
-    # Non-voltage channels are dropped by the export function.
-    # Note: we can write these other channels, such as 'misc'
-    # but these are simply a "catch all" for unknown or undesired
-    # channels.
-    voltage_types = list(units) + ["stim", "misc"]
-    non_voltage_ch = [ch not in voltage_types for ch in orig_ch_types]
-    if any(non_voltage_ch):
-        warn(
-            f"Non-voltage channels detected: {non_voltage_ch}. MNE-Python's "
-            "EDF exporter only supports voltage-based channels, because the "
-            "EDF format cannot accommodate much of the accompanying data "
-            "necessary for channel types like MEG and fNIRS (channel "
-            "orientations, coordinate frame transforms, etc). You can "
-            "override this restriction by setting those channel types to "
-            '"misc" but no guarantees are made of the fidelity of that '
-            "approach."
-        )
```

Comment:
Because that warning doesn't make sense to me. I know it's not specific to STIM channels, but EDF can accommodate arbitrary signal types. Just because some domain-specific meta-information cannot be stored is not really an argument against non-voltage channels (especially since we don't store e.g. channel locations for EEG data either). Also, converting to "misc" was not necessary as the warning suggests, so it is misleading.

Reply:
> we don't store e.g. channel locations for EEG data either

oh wow, OK. I'd forgotten that. I just looked at the docstring of `export_raw()` and it does already have appropriate warnings in the `Notes` section (though they do say "stim channels will be dropped" --- you should change that in this PR).

Reply:
Done!